### PR TITLE
Add AdventJS. A spanish javascript problemset website by midudev

### PR DESCRIPTION
**Pull request by davorpa**
## What does this PR do?
Add resource(s)

### Description

Add AdventJS. A spanish javascript problemset website by midudev

### Why is this valuable (or not)?

Since 2015, the Advent Of Code has been celebrated, which brings programming challenges every day. In this case, Miguel Ángel Durán @midudev brings us challenges to be solved with JavaScript.

### How do we know it's really free?

image

### For book lists, is it a book? For course lists, is it a course? etc.

A competitive and problem sets website available each year in December between 1-25. It's also posible solve problems outside that dates.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)
- [ ] Search for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Review by LuigiImVector — APPROVED** [review]
Great website!

File: more/problem-sets-competitive-programming.md [hunk 1]
```diff
@@ -84,6 +84,7 @@
 * [500 Data structures and algorithms interview questions and their solutions in C++](https://www.quora.com/q/techiedelight/500-Data-Structures-and-Algorithms-interview-questions-and-their-solutions)
 * [A2 Online Judge](https://a2oj.com/ps)
 * [Advent Of Code](http://adventofcode.com)
+* [AdventJS - 25 días de retos con JavaScript](https://adventjs.dev) - Miguel Ángel Durán «midudev» *(GitHub account requested, not required)*
```
Comment (LuigiImVector) [inline more/problem-sets-competitive-programming.md:87]:
```suggestion
* [AdventJS - 25 días de retos con JavaScript](https://adventjs.dev) - Miguel Ángel Durán, midudev *(GitHub account requested, not required)*
```

"«midudev»" is a bit ugly, better to write it this way in my opinion

Comment (davorpa) [reply more/problem-sets-competitive-programming.md:87]: midudev is his nickname

Comment (LuigiImVector) [reply more/problem-sets-competitive-programming.md:87]: I do not like the angular brackets so much, let's wait for another opinion

Comment (eshellman) [reply more/problem-sets-competitive-programming.md:87]: If I understand this correctly I would just omit the attribution. Looks to me like midudev is an organizer not really an author.

Comment (davorpa) [reply more/problem-sets-competitive-programming.md:87]: > If I understand this correctly I would just omit the attribution. Looks to me like midudev is an organizer not really an author.

Is the organizer and the developer at same time :wink: Also a GitHub Star: https://stars.github.com/profiles/midudev/

**Pull request merged**